Common/src/main/java/at/petrak/hexcasting/api/casting/eval/CastingEnvironment.java

```diff
@@ -42,21 +43,21 @@ public abstract class CastingEnvironment {
     /**
      * Stores all listeners that should be notified whenever a CastingEnvironment is initialised.
      */
-    private static final List<Consumer<CastingEnvironment>> createEventListeners = new ArrayList<>();
+    private static final List<BiConsumer<CastingEnvironment, CompoundTag>> createEventListeners = new ArrayList<>();
 
     /**
      * Add a listener that will be called whenever a new CastingEnvironment is created.
      */
-    public static void addCreateEventListener(Consumer<CastingEnvironment> listener) {
```

Comment:
Would it be worth adding an overload for the old signature? something like `createEventListeners.add((env, userData) ->listener.invoke(env))` (syntax??)